File: lilac/src/main/java/com/hartwig/hmftools/lilac/sam/Indel.java
package com.hartwig.hmftools.lilac.sam;

public class Indel
{
    public final boolean IsInsert;
    public final boolean IsDelete;
    public final int Length;
    public final String Contig;
    public final int Position;
    public final String Ref;
    public final String Alt;

    public Indel(final String contig, int position, final String ref, final String alt)
    {
        Contig = contig;
        Position = position;
        Ref = ref;
        Alt = alt;
        IsInsert = Ref.length() < Alt.length();
        IsDelete = !IsInsert;
        Length = Alt.length() - Ref.length();
    }

    public static Indel fromString(final String line)
    {
        final String[] items = line.split("\t");
        return new Indel(items[0], Integer.parseInt(items[1]), items[2], items[3]);
    }

    public String toString()
    {
        return Contig + ':' + Position + ' ' + Ref + '>' + Alt;
    }

    public final boolean match(final Indel other)
    {
        if(!Contig.equals(other.Contig))
            return false;

        if(Position != other.Position)
            return false;

        if(Ref.length() != other.Ref.length())
            return false;

        if(Alt.length() != other.Alt.length())
            return false;

        if(Alt.length() > Ref.length())
        {
            // check the insert
            if(!Alt.substring(1).equals(other.Alt.substring(1)))
                return false;
        }

        return true;
    }
}

File: lilac/src/main/kotlin/com/hartwig/hmftools/lilac/sam/Indel.kt
package com.hartwig.hmftools.lilac.sam

data class Indel(val contig: String, val position: Int, val ref: String, val alt: String) {

    companion object {
        fun fromString(line: String): Indel {
            val (contig, position, ref, alt) = line.split("\t")
            return Indel(contig, position.toInt(), ref, alt)
        }
    }

    val isInsert = ref.length < alt.length
    val isDelete = !isInsert
    val length = alt.length - ref.length
    override fun toString(): String {
        return "$contig:$position $ref>$alt"
    }
}